### PR TITLE
Improve Docstrings in `aepsych/models`

### PR DESCRIPTION
**Pull request by yalsaffar**
Improves documentation in `aepsych/models` for better clarity and consistency.

- Adds missing docstrings to functions and methods across all models.
- Updates existing docstrings with refined type hints and a unified structure.

### CODE REVIEW

File: aepsych/models/pairwise_probit.py [hunk 1]
```diff
@@ -99,6 +119,13 @@ def fit(
         optimizer_kwargs: Optional[Dict[str, Any]] = None,
         **kwargs,
     ) -> None:
+        """Fit the model to the training data.
+
+        Args:
+            train_x (torch.Tensor): Trainin x points.
+            train_y (torch.Tensor): Training y points.
+            optimizer_kwargs (Optional[Dict[str, Any]], optional): Keyword arguments to pass to the optimizer. Defaults to None.
+        """
```
Comment (JasonKChow) [inline aepsych/models/pairwise_probit.py:128]: Are the **kwargs actually useful? If not, it should be removed from the function signature.

Comment (yalsaffar) [reply aepsych/models/pairwise_probit.py:128]: This is a functional change, and I'm not sure if it is useful or not. I'll need to test it and potentially implement this adjustment in a separate PR.

**Review by JasonKChow — APPROVED** [review]
Need rebase.

File: aepsych/models/base.py [hunk 2]
```diff
@@ -310,13 +320,21 @@ def dim_grid(
...
+            slice_dims (Optional[Mapping[int, float]], optional): Dimensions to fix at a 
+            certain value. Defaults to None."""
         return dim_grid(self.lb, self.ub, gridsize, slice_dims)
 
     def set_train_data(self, inputs: Optional[torch.Tensor] = None, targets: Optional[torch.Tensor] = None, strict: bool = False):
         """
-        :param torch.Tensor inputs: The new training inputs.
-        :param torch.Tensor targets: The new training targets.
-        :param bool strict: (default False, ignored). Here for compatibility with
+        Args:
```
[the remark this answers is not in the document]
Comment (JasonKChow) [reply aepsych/models/base.py:343]: Please double check.

File: aepsych/models/gp_classification.py [hunk 3]
```diff
@@ -259,7 +260,7 @@ def sample(self, x: torch.Tensor, num_samples: int) -> torch.Tensor:
 
         Args:
             x (torch.Tensor): Points at which to sample.
-            num_samples (int, optional): Number of samples to return. Defaults to None.
+            num_samples (int): Number of samples to return. Defaults to None.
```
Comment (JasonKChow) [inline aepsych/models/gp_classification.py:263]: Doesn't default to None.

File: aepsych/models/utils.py [hunk 4]
```diff
@@ -237,14 +270,37 @@ class TargetDistancePosteriorTransform(PosteriorTransform):
     def __init__(
         self, target_value: Union[float, Tensor], weights: Optional[Tensor] = None
     ) -> None:
+        """Initialize the TargetDistancePosteriorTransform
+        
+        Args:
+            target_value (Union[float, Tensor]): The target value to transform the posterior to.
+            weights (Tensor, optional): Weights to apply to the target value. Defaults to None.
+        """
```
Comment (JasonKChow) [inline aepsych/models/utils.py:278]: Tensor vs torch.Tensor needs to be consistent, at least within a file.

**Review by JasonKChow — CHANGES_REQUESTED** [review]
Small changes and we'll be ready.

File: aepsych/models/utils.py (hunk 4 above)
Comment (yalsaffar) [reply aepsych/models/utils.py:277]: Made all `models` files consistent using `torch.Tensor`. 
Let me know if I missed anything.

**Review by JasonKChow — CHANGES_REQUESTED** [review]
Need rebase

**Review by JasonKChow — APPROVED** [review]

**Review by JasonKChow — CHANGES_REQUESTED** [review]
Rebase then good.

**Review by JasonKChow — APPROVED** [review]

File: aepsych/models/pairwise_probit.py [hunk 5]
```diff
@@ -157,13 +201,34 @@ def predict_probability(
...
+            Tuple[torch.Tensor, torch.Tensor]: Posterior mean and variance at queries points.
+        """
         return self.predict(
             x, probability_space=True, num_samples=num_samples, rereference=rereference
         )
 
     def sample(
         self, x: torch.Tensor, num_samples: int, rereference: str = "x_min"
     ) -> torch.Tensor:
+        """Sample from the model.
```
Comment (crasanders) [inline aepsych/models/pairwise_probit.py:222]: This change should be made for all the `sample` methods.

```suggestion
        """Sample from the model model posterior.
```

**Review by crasanders — CHANGES_REQUESTED** [review]
Just make that small change to the `sample` docstrings, and I think this is ready.

**Pull request closed without merging**